drivers/display/ssd1322.c

```diff
@@ -0,0 +1,275 @@
+/*
+ * Copyright (c) 2024 Lukasz Hawrylko
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT solomon_ssd1322
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ssd1322, CONFIG_DISPLAY_LOG_LEVEL);
+
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/drivers/display.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/mipi_dbi.h>
+#include <zephyr/kernel.h>
+
+#include "ssd1322_regs.h"
+
+struct ssd1322_config {
+	const struct device *mipi_dev;
+	struct mipi_dbi_config dbi_config;
+	uint16_t height;
+	uint16_t width;
+	uint16_t column_offset;
+};
+
+static inline int ssd1322_write_command(const struct device *dev, uint8_t cmd,
+					const uint8_t *buf, size_t len)
+{
+	const struct ssd1322_config *config = dev->config;
+
+	return mipi_dbi_command_write(config->mipi_dev, &config->dbi_config, cmd, buf, len);
+}
+
+static inline int ssd1322_write_data(const struct device *dev, uint8_t *buf, size_t len)
+{
+	const struct ssd1322_config *config = dev->config;
+	struct display_buffer_descriptor mipi_desc;
+
+	mipi_desc.buf_size = len;
```

Comment:
You should propagate the `height` and `width` here as well- some controllers depend on these values being set to determine how much data to send.

Reply:
Because of the 1 bit -> 4bit conversion and sending data in chunks I cannot propagate width and height that I am receiving from upper layer. I hope that current solution will be acceptable :)